packages/components/pie-modal/src/index.ts

```diff
@@ -80,25 +98,23 @@ export class PieModal extends RtlMixin(LitElement) {
         this.isOpen = false;
     };
 
-    connectedCallback () : void {
-        super.connectedCallback();
-        document.addEventListener(ON_MODAL_OPEN_EVENT, this._handleModalOpened.bind(this));
-        document.addEventListener(ON_MODAL_CLOSE_EVENT, this._handleModalClosed.bind(this));
-    }
-
-    disconnectedCallback () : void {
-        document.removeEventListener(ON_MODAL_OPEN_EVENT, this._handleModalOpened.bind(this));
-        document.removeEventListener(ON_MODAL_CLOSE_EVENT, this._handleModalClosed.bind(this));
-        super.disconnectedCallback();
-    }
+    /**
+     * Prevents the user from dismissing the dialog via the `cancel`
+     * event (ESC key) when `isDismissible` is set to false.
+     *
+     * @param event
+     */
+    private _handleDialogCancelEvent = (event: Event) : void => {
+        if (!this.isDismissible) event.preventDefault();
```

Comment:
NIT: please can this use brackets and separate lines? I think single line `if` statements are kind of difficult to absorb for some reason

Reply:
I'm not 100% sure, but our linting rules may force this atm. Worth checking as can't completely remember.

Reply:
Personally i think this reads better but have updated 👍